runtime/timelib_wrapper.cpp

```diff
@@ -215,3 +225,511 @@ std::pair<int64_t, bool> php_timelib_strtotime(const string &tz_name, const stri
+                               static_cast<int>(t->h), static_cast<int>(t->i), static_cast<int>(t->s), localtime ? ((offset->offset < 0) ? '-' : '+') : '+',
+                               localtime ? abs(offset->offset / 3600) : 0, localtime ? abs((offset->offset % 3600) / 60) : 0);
+        break;
+      case 'r':
+        length = safe_snprintf(buffer, "%3s, %02d %3s %04ld %02d:%02d:%02d %c%02d%02d", php_date_short_day_name(t->y, t->m, t->d), static_cast<int>(t->d),
+                               PHP_TIMELIB_MON_SHORT_NAMES[t->m - 1], static_cast<int64_t>(t->y), static_cast<int>(t->h), static_cast<int>(t->i),
+                               static_cast<int>(t->s), localtime ? ((offset->offset < 0) ? '-' : '+') : '+', localtime ? abs(offset->offset / 3600) : 0,
+                               localtime ? abs((offset->offset % 3600) / 60) : 0);
+        break;
+      case 'U':
+        length = safe_snprintf(buffer, "%lld", t->sse);
+        break;
+
+      case '\\':
+        if (i < format.size()) {
+          ++i;
+        }
+        [[fallthrough]];
+
+      default:
+        buffer[0] = format[i];
+        buffer[1] = '\0';
+        length = 1;
+        break;
+    }
+    SB.append(buffer.data(), length);
+  }
+
+  return SB.str();
+}
+
+string php_timelib_date_format_localtime(const string &format, timelib_time *t) {
+  return php_timelib_date_format(format, t, t->is_localtime);
+}
+
+void php_timelib_date_timestamp_set(timelib_time *t, int64_t timestamp) {
+  auto script_guard = make_malloc_replacement_with_script_allocator();
+  timelib_unixtime2local(t, static_cast<timelib_sll>(timestamp));
+  timelib_update_ts(t, nullptr);
+  t->us = 0;
+}
+
+int64_t php_timelib_date_timestamp_get(timelib_time *t) {
+  auto script_guard = make_malloc_replacement_with_script_allocator();
+
+  timelib_update_ts(t, nullptr);
+
+  int error = 0;
+  timelib_long timestamp = timelib_date_to_int(t, &error);
+  // the 'error' should be always 0 on x64 platform
+  php_assert(error == 0);
+
+  return timestamp;
+}
+
+std::pair<bool, string> php_timelib_date_modify(timelib_time *t, const string &modifier) {
+  auto script_guard = make_malloc_replacement_with_script_allocator();
+
+  timelib_error_container *err = nullptr;
+  timelib_time *tmp_time = timelib_strtotime(modifier.c_str(), modifier.size(), &err, timelib_builtin_db(), timelib_parse_tzfile);
+  vk::final_action tmp_time_deleter{[tmp_time] { timelib_time_dtor(tmp_time); }};
+
+  /* update last errors and warnings */
+  update_errors_warnings(err, script_guard);
+
+  if (err && err->error_count) {
+    /* spit out the first library error message, at least */
+    return {false, gen_parse_error_msg(*err, modifier)};
+  }
+
+  std::memcpy(&t->relative, &tmp_time->relative, sizeof(timelib_rel_time));
+  t->have_relative = tmp_time->have_relative;
+  t->sse_uptodate = 0;
+
+  if (tmp_time->y != -99999) {
```

Comment:
Maybe add a named const for `-99999`?
It's declared in timelib as follows:
```
#define TIMELIB_UNSET   -99999
```

Reply:
fixed